core/chain_util.go

```diff
@@ -346,3 +349,34 @@ func GetBlockByHashOld(db ethdb.Database, hash common.Hash) *types.Block {
 	}
 	return (*types.Block)(&block)
 }
+
+// returns a formatted MIP mapped key.
+func mipmapKey(num, level uint64) []byte {
+	key := new(big.Int).SetUint64(num / level * level)
+
+	return append(mipmapPre, key.Bytes()...)
+}
```

Comment:
Shouldn't the mipmap key contain the level too? Otherwise you are essentially saturating the same (5 in total bloom filters) for the entire system.

E.g.

`mipmapKey(n, 1000)` will be `0` for all `n = 1, 999]`, but
`mipmapKey(n, 1000000)` will also be `0` for all `n = [1, 999999]`
so you are essentially shoving the 1M level into the 1K mipmap too.

Shouldn't the various mipmap levels be completely isolated from one another?